pkg/controller/datavolume-controller.go

```diff
@@ -2599,3 +2623,288 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+	if err != nil {
+		return nil
+	}
+	// Generate individual specs
+	container := makeSizeDetectionContainerSpec(image, pullPolicy)
+	volume := makeSizeDetectionVolumeSpec(sourcePvc.Name)
+	objectMeta := makeSizeDetectionObjectMeta(sourcePvc, true, false)
+	// Assemble the pod
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: *objectMeta,
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				*container,
+			},
+			Volumes: []corev1.Volume{
+				*volume,
+			},
+			RestartPolicy:     corev1.RestartPolicyOnFailure,
+			NodeSelector:      workloadNodePlacement.NodeSelector,
+			Tolerations:       workloadNodePlacement.Tolerations,
+			Affinity:          workloadNodePlacement.Affinity,
+			PriorityClassName: getPriorityClass(sourcePvc),
+		},
+	}
+
+	return pod
+}
+
+// makeSizeDetectionObjectMeta creates and returns the object metadata for the size-detection pod
+func makeSizeDetectionObjectMeta(sourcePvc *corev1.PersistentVolumeClaim, blockOwnerDeletion, isController bool) *metav1.ObjectMeta {
+	return &metav1.ObjectMeta{
+		Name:      sizeDetectionPodName(sourcePvc),
+		Namespace: sourcePvc.Namespace,
+		Annotations: map[string]string{
+			AnnCreatedBy: "yes",
+		},
+		Labels: map[string]string{
+			common.CDILabelKey:        common.CDILabelValue,
+			common.CDIComponentLabel:  common.ImporterPodName,
+			common.PrometheusLabelKey: common.PrometheusLabelValue,
+		},
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				APIVersion:         "v1",
+				Kind:               "PersistentVolumeClaim",
+				Name:               sourcePvc.Name,
+				UID:                sourcePvc.GetUID(),
+				BlockOwnerDeletion: &blockOwnerDeletion,
+				Controller:         &isController,
+			},
+		},
+	}
+}
+
+// makeSizeDetectionContainerSpec creates and returns the size-detection pod's Container spec
+func makeSizeDetectionContainerSpec(image, pullPolicy string) *corev1.Container {
+	return &corev1.Container{
+		Name:            "size-detection-volume",
+		Image:           image,
+		ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+		Command:         []string{"/usr/bin/cdi-image-size-detection"},
+		Args:            []string{"-url", common.ImporterWritePath, "-path", corev1.TerminationMessagePathDefault},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: common.ImporterDataDir,
+				Name:      DataVolName,
+			},
+		},
+	}
+}
+
+// makeSizeDetectionVolumeSpec creates and returns the size-detection pod's Volume spec
+func makeSizeDetectionVolumeSpec(pvcName string) *corev1.Volume {
+	return &corev1.Volume{
+		Name: DataVolName,
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvcName,
+				ReadOnly:  false,
+			},
+		},
+	}
+}
+
+// inflateSizeWithOverhead inflates the obtained image size with proper overhead calculations
+func inflateSizeWithOverhead(client client.Client,
+	imgSize string,
```

Comment:
I think this being either a quantity or int64 would be better, that way we parse at the source, and pass it along in the right type format. Then you don't have to parse it in this function.